devito/ir/support/utils.py

```diff
@@ -77,7 +77,7 @@ def build_iterators(mapper):
     iterators = OrderedDict()
     for k, v in mapper.items():
         for d, offs in v.items():
-            if d.is_Stepping:
+            if d.is_Stepping and k is not None:
```

Comment:
not sure why ``k`` could be None ( I don't remember), but even then, I wonder whether we should rather have a
```
if k is None:
    continue
```
right below the ``for k, v ...`` loop?

Reply:
`k` could be None due to the new ~Add,..` than have empty stuff in it in some cases

Reply:
the keys in ``mapper`` are expected to be pure ``Function/TimeFunction/...``. We should have no ``Add/...`` at this stage. If we have them, that should be fixed.